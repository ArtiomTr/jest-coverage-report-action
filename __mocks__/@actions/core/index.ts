const input: Record<string, unknown> = {};

export const getInput = (name: string) => input[name];

export const mockInput = (newInput: Record<string, unknown>) => {
    Object.assign(input, newInput);
};

export const clearInputMock = () => {
    Object.keys(input).map((key) => {
        delete input[key];
    });
};

export const error = jest.fn();
export const info = jest.fn();
export const setFailed = jest.fn();
