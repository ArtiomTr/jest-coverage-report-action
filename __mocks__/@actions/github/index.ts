export const context: Record<string, unknown> = {};

export const mockContext = (newContext: Record<string, unknown>) => {
    Object.assign(context, newContext);
};

export const clearContextMock = () => {
    Object.keys(context).map((key) => {
        delete context[key];
    });
};

export const getOctokit = jest.fn();
