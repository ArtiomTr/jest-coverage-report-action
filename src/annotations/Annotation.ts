import type { operations } from '@octokit/openapi-types';

export type Annotation = Required<
    Required<
        operations['checks/create']['requestBody']['content']['application/json']
    >['output']
>['annotations'][number];
