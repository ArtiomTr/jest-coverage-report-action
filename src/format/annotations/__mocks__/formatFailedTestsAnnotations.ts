export const formatFailedTestsAnnotations = jest.fn();
