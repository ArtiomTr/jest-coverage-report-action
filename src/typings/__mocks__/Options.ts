export const getOptions = jest.fn(() => ({
    token: 'TOKEN',
    testScript: 'test script',
    iconType: 'emoji',
    annotations: 'all',
    threshold: 80,
    packageManager: 'npm',
    skipStep: 'none',
}));
