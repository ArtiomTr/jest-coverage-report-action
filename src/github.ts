import { context, getOctokit } from '@actions/github';

import {
    PullRequestInfo,
    PullRequestInfoQuery,
    PullRequestInfoQueryVariables,
} from './generated/graphql';

export type Octokit = ReturnType<typeof getOctokit>;

function graphql<TQuery, TQueryVariables extends Record<string, unknown>>(
    octokit: Octokit,
    source: string,
    variables: TQueryVariables
) {
    return octokit.graphql<TQuery>(source, {
        ...variables,
    });
}

export const getPullRequestInfo = async (
    octokit: Octokit,
    prNumber?: number
) => {
    if (context?.payload?.pull_request) {
        return context?.payload?.pull_request;
    }

    if (!prNumber) {
        return;
    }

    const output = await graphql<
        PullRequestInfoQuery,
        PullRequestInfoQueryVariables
    >(octokit, PullRequestInfo, {
        repo: context.repo.repo,
        owner: context.repo.owner,
        prNumber,
    });

    output.repository?.pullRequest?.baseRef?.repository.nameWithOwner;
};
