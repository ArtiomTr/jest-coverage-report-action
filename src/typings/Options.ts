import { getInput } from '@actions/core';
import { context, getOctokit } from '@actions/github';
import * as yup from 'yup';

import { icons } from '../format/strings.json';

export type IconType = keyof typeof icons;

export type AnnotationType = 'all' | 'none' | 'coverage' | 'failed-tests';
export type PackageManagerType = 'npm' | 'yarn' | 'pnpm';
export type SkipStepType = 'all' | 'none' | 'install';
export type PullRequest = {
    base: { ref: string };
    head: { ref: string; sha: string };
    number: number;
};
export type Options = {
    token: string;
    testScript: string;
    iconType: IconType;
    annotations: AnnotationType;
    threshold?: number;
    workingDirectory?: string;
    packageManager: PackageManagerType;
    skipStep: SkipStepType;
    customTitle?: string;
    coverageFile?: string;
    baseCoverageFile?: string;
    prNumber: null | number;
    pullRequest: null | PullRequest;
};

const validAnnotationOptions: Array<AnnotationType> = [
    'all',
    'none',
    'coverage',
    'failed-tests',
];

const packageManagerOptions: Array<PackageManagerType> = [
    'npm',
    'yarn',
    'pnpm',
];

const validIconOptions = Object.keys(icons);

const validSkipStepOptions: Array<SkipStepType> = ['all', 'none', 'install'];

const optionSchema = yup.object().shape({
    token: yup.string().required(),
    testScript: yup.string().required(),
    iconType: yup.string().required().oneOf(validIconOptions),
    annotations: yup.string().required().oneOf(validAnnotationOptions),
    threshold: yup
        .number()
        .transform((value) => (isNaN(value) ? undefined : value))
        .min(0)
        .max(100),
    workingDirectory: yup.string(),
    packageManager: yup.string().required().oneOf(packageManagerOptions),
    skipStep: yup.string().required().oneOf(validSkipStepOptions),
    customTitle: yup.string(),
    coverageFile: yup.string(),
    baseCoverageFile: yup.string(),
    prNumber: yup.number().nullable(),
    pullRequest: yup.object().nullable(),
});

export const shouldInstallDeps = (skipStep: SkipStepType): Boolean =>
    !['all', 'install'].includes(skipStep);

export const shouldRunTestScript = (skipStep: SkipStepType): Boolean =>
    !['all'].includes(skipStep);

export const getOptions = async (): Promise<Options> => {
    const token = getInput('github-token', {
        required: true,
    });
    const octokit = getOctokit(token);
    const testScript = getInput('test-script');
    const threshold = getInput('threshold');
    const workingDirectory = getInput('working-directory');
    const iconType = getInput('icons');
    const annotations = getInput('annotations');
    const packageManager = getInput('package-manager');
    const skipStep = getInput('skip-step');
    const customTitle = getInput('custom-title');
    const coverageFile = getInput('coverage-file');
    const baseCoverageFile = getInput('base-coverage-file');
    const prNumber: number | null = Number(
        getInput('prnumber') || context?.payload?.pull_request?.number
    );
    let pullRequest = context?.payload?.pull_request || null;

    if (!pullRequest && !Number.isNaN(prNumber)) {
        const { data: pr } = await octokit.pulls.get({
            owner: context.repo.owner,
            repo: context.repo.repo,
            pull_number: prNumber,
        });
        pullRequest = pr as PullRequest;
    }

    try {
        const options: Options = (await optionSchema.validate({
            token,
            testScript,
            threshold,
            workingDirectory,
            iconType,
            annotations,
            packageManager,
            skipStep,
            customTitle,
            coverageFile,
            baseCoverageFile,
            prNumber: prNumber || null,
            pullRequest,
        })) as Options;

        return options;
    } catch (err) {
        if (err instanceof yup.ValidationError) {
            throw new Error(
                [err.message, ...err.errors].filter(Boolean).join('\n')
            );
        }

        throw err;
    }
};
