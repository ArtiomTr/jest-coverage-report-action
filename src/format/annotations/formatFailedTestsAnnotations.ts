import { context } from '@actions/github';

import { CreateCheckOptions } from './CreateCheckOptions';
import { getFailedTestsAnnotationsBody } from './getFailedTestsAnnotationsBody';
import { Annotation } from '../../annotations/Annotation';
import { Options } from '../../typings/Options';
import { TestRunReport } from '../../typings/Report';
import { i18n } from '../../utils/i18n';

export const formatFailedTestsAnnotations = (
    runReport: TestRunReport,
    annotations: Array<Annotation>,
    options: Options
): CreateCheckOptions => ({
    ...context.repo,
    status: 'completed',
    head_sha: options?.pullRequest?.head?.sha ?? context.sha,
    conclusion: 'failure',
    name: i18n('failedTestsCheckName'),
    output: {
        title: i18n('testsFail'),
        text: [
            getFailedTestsAnnotationsBody(runReport),
            annotations.length > 50 &&
                i18n('tooMuchAnnotations', {
                    hiddenCount: annotations.length - 50,
                }),
        ]
            .filter(Boolean)
            .join('\n'),
        summary: runReport.summary,
        annotations: annotations.slice(0, 49),
    },
});
