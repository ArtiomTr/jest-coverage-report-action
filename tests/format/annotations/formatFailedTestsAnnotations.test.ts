import * as all from '@actions/github';

import { Annotation } from '../../../src/annotations/Annotation';
import { formatFailedTestsAnnotations } from '../../../src/format/annotations/formatFailedTestsAnnotations';
import { Options } from '../../../src/typings/Options';

const { mockContext, clearContextMock } = all as any;

const DEFAULT_OPTIONS: Options = {
    token: '',
    testScript: '',
    iconType: 'emoji',
    annotations: 'all',
    packageManager: 'npm',
    skipStep: 'all',
    prNumber: 5,
    pullRequest: {
        number: 5,
        head: { sha: '987654', ref: '123' },
        base: { ref: '456' },
    },
    output: ['comment'],
};

const annotations: Annotation[] = [
    {
        annotation_level: 'failure',
        path: 'tests/index.test.ts',
        start_line: 30,
        end_line: 25,
        title: 'test > lol > a',
        message: 'Test failed...',
    },
    {
        annotation_level: 'failure',
        path: 'tests/sendMessage.test.ts',
        start_line: 11,
        end_line: 11,
        title: 'test > lol > a',
        message: 'Test failed...',
    },
];

beforeEach(clearContextMock);

describe('formatFailedTestsAnnotations', () => {
    it('should format failed tests annotations for PR', () => {
        mockContext({
            repo: {
                owner: 'test-bot',
                repo: 'test',
            },
            payload: {
                pull_request: {
                    head: {
                        sha: '987654',
                    },
                },
            },
        });

        expect(
            formatFailedTestsAnnotations(
                {
                    title: 'Test success',
                    summary: 'Some summary',
                    failures: 'Failures',
                },
                annotations,
                DEFAULT_OPTIONS
            )
        ).toMatchSnapshot();
    });

    it('should format failed tests annotations for commit', () => {
        mockContext({
            repo: {
                owner: 'test-bot',
                repo: 'test',
            },
            payload: {},
            sha: 'as12d1',
        });

        expect(
            formatFailedTestsAnnotations(
                {
                    title: 'Test success',
                    summary: 'Some summary',
                    failures: 'Failures',
                },
                annotations,
                { ...DEFAULT_OPTIONS, prNumber: null, pullRequest: null }
            )
        ).toMatchSnapshot();
    });

    it('should return only 50 annotations', () => {
        mockContext({
            repo: {
                owner: 'test-bot',
                repo: 'test',
            },
            payload: {},
            sha: 'as12d1',
        });

        expect(
            formatFailedTestsAnnotations(
                {
                    title: 'Test success',
                    summary: 'Some summary',
                    failures: 'Failures',
                },
                new Array(53).fill(0).map(() => ({ ...annotations[0] })),
                { ...DEFAULT_OPTIONS, prNumber: null, pullRequest: null }
            )
        ).toMatchSnapshot();
    });
});
