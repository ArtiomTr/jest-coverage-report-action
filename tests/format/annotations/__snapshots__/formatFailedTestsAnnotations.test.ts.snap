// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formatFailedTestsAnnotations > should format failed tests annotations for PR 1`] = `
{
  "conclusion": "failure",
  "head_sha": "987654",
  "name": "Tests annotations (ðŸ§ª jest-coverage-report-action)",
  "output": {
    "annotations": [
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 11,
        "message": "Test failed...",
        "path": "tests/sendMessage.test.ts",
        "start_line": 11,
        "title": "test > lol > a",
      },
    ],
    "summary": "Some summary",
    "text": "Created failed tests' annotations. To disable them, see [documentation](https://github.com/ArtiomTr/jest-coverage-report-action#jest-coverage-report-).
Failures",
    "title": "Test suite run failed",
  },
  "owner": "test-bot",
  "repo": "test",
  "status": "completed",
}
`;

exports[`formatFailedTestsAnnotations > should format failed tests annotations for commit 1`] = `
{
  "conclusion": "failure",
  "head_sha": "as12d1",
  "name": "Tests annotations (ðŸ§ª jest-coverage-report-action)",
  "output": {
    "annotations": [
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 11,
        "message": "Test failed...",
        "path": "tests/sendMessage.test.ts",
        "start_line": 11,
        "title": "test > lol > a",
      },
    ],
    "summary": "Some summary",
    "text": "Created failed tests' annotations. To disable them, see [documentation](https://github.com/ArtiomTr/jest-coverage-report-action#jest-coverage-report-).
Failures",
    "title": "Test suite run failed",
  },
  "owner": "test-bot",
  "repo": "test",
  "status": "completed",
}
`;

exports[`formatFailedTestsAnnotations > should return only 50 annotations 1`] = `
{
  "conclusion": "failure",
  "head_sha": "as12d1",
  "name": "Tests annotations (ðŸ§ª jest-coverage-report-action)",
  "output": {
    "annotations": [
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
      {
        "annotation_level": "failure",
        "end_line": 25,
        "message": "Test failed...",
        "path": "tests/index.test.ts",
        "start_line": 30,
        "title": "test > lol > a",
      },
    ],
    "summary": "Some summary",
    "text": "Created failed tests' annotations. To disable them, see [documentation](https://github.com/ArtiomTr/jest-coverage-report-action#jest-coverage-report-).
Failures
3 annotations hidden. Only 50 can be displayed at once.",
    "title": "Test suite run failed",
  },
  "owner": "test-bot",
  "repo": "test",
  "status": "completed",
}
`;
